src/DssVest.sol

```diff
@@ -174,6 +176,48 @@ abstract contract DssVest is ERC2771Context {
         require((z = uint128(x)) == x, "DssVest/uint128-overflow");
     }
 
+    /** 
+        @dev commit to the creation of an award without revealing the award's contents yet
+        @param bch  Blind Commitment Hash - The hash of the award's contents, see hash function in createAward for details
+    */
+    function commit(bytes32 bch) external lock auth {
+        commitments[bch] = true;
+    }
+
+    /**
+        @dev Create a vesting contract from an earlier commitment
+        @param _bch The hash of the award's contents, see hash function in createAward for details
+        @param _usr The recipient of the reward
+        @param _tot The total amount of the vest
+        @param _bgn The starting timestamp of the vest
+        @param _tau The duration of the vest (in seconds)
+        @param _eta The cliff duration in seconds (i.e. 1 years)
+        @param _mgr An optional manager for the contract. Can yank if vesting ends prematurely.
+        @param _slt The salt used to increase privacy when committing
+        @return id  The id of the vesting contract
+    */
+    function createFromCommitment(bytes32 _bch, address _usr, uint256 _tot, uint256 _bgn, uint256 _tau, uint256 _eta, address _mgr, bytes32 _slt) external lock returns (uint256 id) {
+        require(_bch == keccak256(abi.encodePacked(_usr, _tot, _bgn, _tau, _eta, _mgr, _slt)), "DssVest/invalid-hash");
+        require(commitments[_bch], "DssVest/commitment-not-found");
+        id = _create(_usr, _tot, _bgn, _tau, _eta, _mgr);
+        commitments[_bch] = false;
```

Comment:
it is safer to set `commitments[_bch] = false;` before `create` is executed
, right after it was checked that the commit exists.

Reply:
(this is for protecting against reentrancy, but I know `lock` is already doing that. It is just a second layer of security, technically not needed.

Reply:
done [b282ab2](https://github.com/corpus-io/dss-vest/pull/25/commits/b282ab2866f07ef33632d00eda49747be1bf5842)